sdtw/soft_dtw.py

```diff
@@ -33,6 +37,10 @@ def __init__(self, D, gamma=1.0):
         self.D = self.D.astype(np.float64)
 
         self.gamma = gamma
+        self.sakoe_chiba_band = sakoe_chiba_band
+
+        if sakoe_chiba_band >= 0:
+            assert self.D.shape[0] == self.D.shape[1]
```

Comment:
One last question before merge... Why do you need this restriction?

Reply:
For simplicity.
I updated the code and removed the restriction.